### ISSUE TITLE
Rails 8, Hotwire, Stimulus, no react

### ISSUE THREAD

**Issue by Alagaesia93** [Shopify/shopify-app-template-ruby#122]
Hello Shopify!

With the upcoming release of [Rails 8](https://github.com/rails/rails/milestone/87), I think the default rails template for a Shopify app could deeply change. This is more a topic for discussions, but there is no discussions tab so here I am.

In particular
- Is React really needed? Embedded apps have a full page reload when changing from the app menu, and inside the app it could easily be another page (index.html.erb to show.html.erb), especially with Hotwire. A full stack dev could just get some Polaris designs and create a fully working app without needing to know one line of React
- If needed, Stimulus could be used for interactivity, replacing the need for React
- Kamal will sit on top of Docker / Kubernetes, allowing to deploy everywhere
- solid queue and solid cache examples (especially on rate limiting Shopify's APIs) => https://github.com/basecamp/solid_queue/issues/97

Benefits
- zero bundling and transpiling - way faster deployments (and easy to debug)
- way faster development cycle
- shorter learning curve
- frontend developers focused on building extensions (admin, storefront, account, checkout etc)

I am very interested in building something like this, I have already seen a few resources online (i.e. https://github.com/kirillplatonov/shopify-hotwire-sample) from @kirillplatonov. 

Is it interesting for Shopify as well? I'm asking because I can see an outdated Ruby version on this repo, for example, and official documentation mentioning only Remix templates.

**Comment by bifo90** [comment]
Yep! I absolutely agree with you. Let's keep a Ruby App template updated 💪

**Comment by jagthedrummer** [comment]
Not stale. Please disable the bot. It's hostile and give the impression that you don't actually want to hear from developers.

**Comment by lizkenyon** [comment]
Hi there 👋 

First of all, I have put up a PR to remove stalebot from this repository. That should have been done months ago when we removed it from other repositories, but some repos were missed, including this one. And sorry for not jumping into this discussion sooner.

We do want to hear the feedback from out community, and thank you for starting this discussion.

Secondly, there has been internal discussions about the future of the ruby template, and what it could look like without React. Including chatting with folks like @kirillplatonov, about their experiences with their template. At this time we don't have anything to announce, but this is a topic the team has been thinking about.

One of the more immediate things we are looking to do is putting together a guide of how to modify the current template without the react frontend.

**Comment by jagthedrummer** [comment]
I'm using Polaris, and I currently link to it on unpkg. (I just noticed that I'm several versions behind, so I don't know if that's even a possibility anymore.)

I pull it into my apps via a gem that I made to help manage both the CSS and the markup.

https://github.com/jagthedrummer/polaris_view_helpers

**Comment by dhmacs** [comment]
I'm primarily a React dev (being using React since 2015) and I've only learned Rails a couple years ago. 

I'd love to see the Shopify Rails template embrace Rails 8 + Hotwire, in fact at my company we've built one in-house because Rails + React template is complicating things that are quite simple in vanilla Rails.

We're currently using [Polaris View Components](https://github.com/baoagency/polaris_view_components), but it'd be great have an Hotwire solution backed by Shopify team. 
It could be styled erb markup + Stimulus controllers to be added where needed

**Comment by danielfriis** [comment]
> ... and we've added a [Wiki page](https://github.com/Shopify/shopify-app-template-ruby/wiki/Replacing-React-frontend-with-Stimulus) on unplugging the React frontend from the current template. Please let us know if you have any issues with it, or if we missed anything!

I followed your guide, but it seems insufficient to get the app fully working. I can't navigate to other controllers or actions without being redirected to the install-page.

**Comment by remy727** [comment]
@danielfriis, just want to make sure you updated `shopify.web.toml` file.

```toml
# shopify.web.toml

roles = ["frontend"]

[commands]
dev = "bin/rails server"
```

**Comment by AAlvarez90** [comment]
I would really love to see this issue addressed. I would like to build a shopify app but using Turbo and Hotwire rather than react. There are ways to do that but it takes some boilerplate code..
